ReactNative/js/globals/window/crypto.ts

```diff
@@ -0,0 +1,101 @@
+import { NativeModules } from 'react-native';
+
+function arrayBufferToHexString(buffer: ArrayBuffer) {
+  return Array.prototype.map
+    .call(new Uint8Array(buffer), x => `00${x.toString(16)}`.slice(-2))
+    .join('');
+}
+
+function hexStringToByteArray(hexString: string): ArrayBuffer {
+  const typedArray = new Uint8Array(
+    (hexString.match(/[\da-f]{2}/gi) || []).map((h: string) => {
+      return parseInt(h, 16);
+    }),
+  );
+  return typedArray.buffer;
+}
+
+function toByteArray(data: any) {
+  if (data.buffer) {
+    return new Uint8Array(data.buffer, data.byteOffset, data.byteLength);
+  }
+  return new Uint8Array(data);
+}
+
+class RandomPool {
+  pool: number[] = [];
+
+  pending: number = 0;
+
+  addEntropy(entropy: ArrayBuffer) {
+    const buffer = new Uint8Array(entropy);
+    Array.prototype.forEach.call(buffer, x => {
+      this.pool.push(x);
+    });
+    this.pending -= buffer.length;
+    if (this.pending < 0) {
+      throw new Error('Something went wrong');
+    }
+  }
+
+  getRandomByte(): number {
+    const randomByte = this.pool.shift();
+    if (this.pool.length + this.pending < 1024 * 48) {
+      this.fetchEntropy(1024 * 64);
```

Comment:
Wwe did not aim for high efficiency here, but in case you notice that it get sluggish, you can safely refetch data is smaller chunks here when you fall below the low watermark.

In the long run, I'm considering moving some parts of the crypto back to JavaScript. For example, hashing, generating random data (still need native code to get entropy sources). However, the benefits would only be performance. Let's see if we need it. Depends also on how expensive the back-and-forth conversion to string and the bridge is.

Candidate library: https://github.com/dchest/tweetnacl-js